### PR TITLE
[14.0] shopfloor, delivery: force qty to process to 1 when scanning a product or a lot

### PR DESCRIPTION
**Pull request by sebalix**
Ref. rau-51

### CODE REVIEW

File: shopfloor/actions/message.py [hunk 1]
```diff
@@ -370,6 +370,15 @@ def product_mixed_package_scan_package(self):
             ),
         }
 
+    def product_not_unitary_in_package_scan_package(self):
+        return {
+            "message_type": "warning",
+            "body": _(
+                "This product is present multiple times in a package, "
```
Comment (jbaudoux) [inline shopfloor/actions/message.py]:
```suggestion
                "This product is part of a package, "
```

File: shopfloor/models/stock_quant_package.py [hunk 2]
```diff
@@ -92,3 +92,7 @@ def _move_package_quant_move_values(self, quant, location, quantity):
             "package_id": quant.package_id.id,
             "owner_id": quant.owner_id.id,
         }
+
+    def has_one_product(self):
+        self.ensure_one()
+        return len(self.quant_ids.product_id) == 1
```
Comment (sebalix) [inline shopfloor/models/stock_quant_package.py]: Note to myself: this method is not used anymore, to remove

**Comment by TDu** [comment]
LG, not sure about that pylint error...

File: shopfloor/actions/message.py (hunk 1 above)
Comment (sebalix) [reply shopfloor/actions/message.py]: fixed

Comment (sebalix) [reply shopfloor/actions/message.py]: fixed

**Review by TDu — APPROVED** [review]

**Comment by simahawk** [comment]
for the pre-commit issue: please open another pr after running `copier -f update`.

**Review by simahawk — APPROVED** [review]
LG

**Pull request merged**